web/src/views/InvestigationDetail.vue

```diff
@@ -44,6 +44,32 @@
             ref="map"
             class="map"
           />
+          <v-menu
+            v-model="externalVisualizationMenu"
+            :position-x="externalVisualizationMenuX"
+            :position-y="externalVisualizationMenuY"
+            absolute
+            offset-y
+            :close-on-click="false"
+          >
+            <v-list>
```

Comment:
I know this addition is relatively small, but this still may be an opportunity to separate this into its own component. I think that `InvestigationDetail.vue` has grown into a monolith that is difficult to navigate simply because of many small additions like this. It may be too late to refactor the whole thing, but I think it's worthwhile to adopt a philosophy of separation as we go forward.

Reply:
I've created a new component for this context menu in 4d49c64